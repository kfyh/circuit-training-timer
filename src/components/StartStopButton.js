import React from 'react'
import PropTypes from 'prop-types'

function StarStopButton ({ isRunning, onClick }) {
  const label = isRunning ? 'Stop' : 'Start'
  return (
    <button onClick={onClick}>
      {label}
    </button>
  )
}

StarStopButton.propTypes = {
  isRunning: PropTypes.bool,
  onClick: PropTypes.func
}

export default StarStopButton
