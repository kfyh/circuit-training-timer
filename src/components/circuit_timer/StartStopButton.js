import React from 'react';
import PropTypes from 'prop-types';

export function StartStopButton({ isRunning, onClick }) {
	const label = isRunning ? 'Stop' : 'Start';
	return <button onClick={onClick}>{label}</button>;
}

StartStopButton.propTypes = {
	isRunning: PropTypes.bool,
	onClick: PropTypes.func,
};
