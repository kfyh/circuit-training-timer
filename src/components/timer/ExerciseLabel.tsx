import React, { ReactElement } from 'react';
import { FlattenedStep } from '../../types/circuits';

type ExerciseLabelProps = {
	step: FlattenedStep;
};

export function ExerciseLabel(props: ExerciseLabelProps): ReactElement {
	const step = props.step;
	const excerciseLabel = step.label;
	const totalReps = step.repTotal;

	return (
		<div>
			{step.parentLabel && (
				<p>
					{step.parentLabel} ({step.parentRepIndex}/{step.parentRepTotal})
				</p>
			)}
			<p>
				{excerciseLabel} ({props.step.repIndex}/{totalReps})
			</p>
		</div>
	);
}
