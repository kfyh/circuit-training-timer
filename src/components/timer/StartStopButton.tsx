import React, { ReactElement } from 'react';

type StartStopButtonProps = {
	isRunning: boolean;
	onClick: () => void;
}

export function StartStopButton({ isRunning, onClick }: StartStopButtonProps): ReactElement {
	const label = isRunning ? 'Stop' : 'Start';
	return <button onClick={onClick}>{label}</button>;
}
