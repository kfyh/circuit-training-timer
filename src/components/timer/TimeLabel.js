import React from 'react';
import PropTypes from 'prop-types';

export function TimeLabel({ time }) {
	const divStyle = {
		color: 'blue',
		fontSize: '250pt',
	};

	let displayTime = Math.ceil(Math.max(0, time / 1000));
	if (displayTime > 60) {
		displayTime = Math.ceil(time / 1000 / 60) + ' mins';
		divStyle.fontSize = '130pt';
	}

	return <div style={divStyle}>{displayTime}</div>;
}

TimeLabel.propTypes = {
	time: PropTypes.number,
};
