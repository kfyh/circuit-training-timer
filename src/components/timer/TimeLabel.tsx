import React, { ReactElement } from 'react';

type TimeLabelProps = {
	time: number;
	isResting: boolean;
};

export function TimeLabel({ time, isResting }: TimeLabelProps): ReactElement {
	const divStyle = {
		color: 'blue',
		fontSize: '250pt',
	};

	if (isResting) {
		divStyle.color = 'green';
	}

	const timeSeconds = Math.ceil(Math.max(0, time / 1000));
	let displayTime = timeSeconds.toString();
	if (timeSeconds > 60) {
		displayTime = Math.ceil(time / 1000 / 60) + ' mins';
		divStyle.fontSize = '130pt';
	}

	return <div style={divStyle}>{displayTime}</div>;
}
