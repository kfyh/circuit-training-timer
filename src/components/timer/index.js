export * from './StartStopButton';
export * from './ExerciseLabel';
export * from './TimeLabel';
