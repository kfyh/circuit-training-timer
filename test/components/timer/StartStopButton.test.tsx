import React from 'react';
import { shallow } from 'enzyme';
import { StartStopButton } from '../../../src/components/timer/StartStopButton';

describe('Given StartStopButton', () => {
	const testValues = [
		[true, 'Stop'],
		[false, 'Start'],
	];

	test.each(testValues)('When isRunning is %b then button label is %s', (value, expected) => {
		const render = shallow(<StartStopButton isRunning={value} />);
		expect(render).toMatchSnapshot();
	});
});
