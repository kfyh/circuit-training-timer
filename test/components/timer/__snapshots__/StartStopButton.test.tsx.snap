// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given StartStopButton When isRunning is %b then button label is false 1`] = `
<button>
  Start
</button>
`;

exports[`Given StartStopButton When isRunning is %b then button label is true 1`] = `
<button>
  Stop
</button>
`;
